### PR TITLE
Fix code generation issue for python

### PR DESCRIPTION
**Pull request by lonewolf3739**
Signed-off-by: Srikanth Chekuri <srikanth.chekuri92@gmail.com>

## Which problem is this PR solving?
Fixes https://github.com/jaegertracing/docker-protobuf/issues/24
Fixes https://github.com/jaegertracing/jaeger-client-csharp/issues/197

## Short description of the changes
**proto-gen-python** can not be used because `google/api/annotations.proto`, `gogoproto/gogo.proto` and `protoc-gen-swagger/options/annotations.proto` are not created but required to use the `model` or `collector` generated files.

Update the Makefile to `generate google/api/annotations.proto`, `gogoproto/gogo.proto` and `protoc-gen-swagger/options/annotations.proto`

### CODE REVIEW

**Comment by Falco20019** [comment]
I would suggest merging `PROTOC_CS_INTERNAL` and `PROTOC_PY_INTERNAL` into `PROTOC_INTERNAL` Since they are called with the same file list. Just add the `python_out` right after the `csharp_out`.

**Review by Falco20019 — CHANGES_REQUESTED** [review]

**Comment by Falco20019** [comment]
Just a small suggestion avoid cluttering the file since the goal and file list is the same for CS and PY.

File: Makefile [hunk 1]
```diff
@@ -111,6 +115,13 @@ proto:
 		protoc-gen-swagger/options/openapiv2.proto \
 		gogoproto/gogo.proto
 
+	$(PROTOC_PY_INTERNAL) \
```
[the remark this answers is not in the document]
Comment (lonewolf3739) [reply Makefile:118]: Yeah, makes sense. I will change it.

[the remark this answers is not in the document]
Comment (Falco20019) [reply Makefile:118]: Or put it before to keep the order similar to `PROTOC_WITHOUT_GRPC`

**Comment by Falco20019** [comment]
You might also want to add a „Fixes https://github.com/jaegertracing/jaeger-client-csharp/issues/197“ to the initial post 👍

File: Makefile [hunk 2]
```diff
@@ -84,9 +84,10 @@ PROTOC_WITH_GRPC := $(PROTOC_WITHOUT_GRPC) \
 		--grpc-cpp_out=${PROTO_GEN_CPP_DIR} \
 		--grpc-csharp_out=${PROTO_GEN_CSHARP_DIR}
 
-PROTOC_CS_INTERNAL := $(PROTOC) \
+PROTOC_INTERNAL := $(PROTOC) \
 		$(PROTO_INCLUDES) \
 		--csharp_out=internal_access,base_namespace:${PROTO_GEN_CSHARP_DIR}
```
[the remark this answers is not in the document]
Comment (lonewolf3739) [reply Makefile]: Updated. I wish it gave error when I ran `make proto` but it ran silently without any error.

**Review by Falco20019 — APPROVED** [review]
LGTM. If travis goes green, I‘m fine with merging

**Comment by Falco20019** [comment]
@yurishkuro I have no write permission, so my review is not worth too much. But looks promising from my side. So if it runs for you, feel free to also give it a look and approve.

**Comment by yurishkuro** [comment]
1. since Travis-CI decided to deny us any credits, we need to add GH Action for this repo (as we already did in many others) to see that the CI build is succeeding
2. it is not clear to me how this is actually fixing the linked issues. The Makefile in this repo is merely a sanity check that `protoc` can run successfully. Are we saying that this Makefile is how we want people to generate stubs? If so, we should make it explicit in the README.
3. But even with (2), we're not actually testing the generated stubs, otherwise we would've discovered this issue with Python. So ideally we'd have some basic test programs for each language that try to import the stubs and do a roundtrip encode-decode (does not need to be a part of this PR).

**Comment by Falco20019** [comment]
I think only the library maintainer would need to do this in most cases. End users usually can just use however the library maintainer for the language decided to package those generated files. Due to the dependency on those five protos, there is no easy way to generate the necessary code other than using this Makefile or by doing the same stuff that the docker does with the wrapper (downloading everything and so on) and then doing the steps in this Makefile.

It might be hard to do the roundtrips only by the generated files. We would still need some bootstrapped per language and maybe some tests like with xdock. But since this is mostly necessary for a small group like 1-2 persons per language, I‘m not sure if it‘s worth the effort.

**Comment by lonewolf3739** [comment]
> Due to the dependency on those five protos, there is no easy way to generate the necessary code other than using this Makefile

Yes, it wasn't clear how to get around with `gogo.proto`, `.../annotations.proto` etc.. I think this will become problem for other language clients as well. As of now C# is the only client that supports sending via gRPC and as I am trying with python I came across this problem. As suggested by @yurishkuro maybe we should recommend generating stubs with `make`?

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**